src/main/java/org/infrastructurebuilder/auditor/PageGenerator.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.maven.doxia.parser.Parser;
+import org.apache.maven.doxia.sink.Sink;
+import org.infrastructurebuilder.auditor.model.AuditResult;
+import org.infrastructurebuilder.auditor.model.AuditorResults;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Named
+public final class PageGenerator {
+  private final Map<String, AuditReporter> reporters;
+  private final static Logger log = LoggerFactory.getLogger(PageGenerator.class);
+
+  @Inject
+  public PageGenerator(Map<String, AuditReporter> reporters) {
+    this.reporters = reporters;
+    log.info(String.format("audit-reporting-maven-plugin found %d reporters", reporters.size()));
+  }
+
+  public void generate(String title, Sink sink) {
+    Objects.requireNonNull(sink);
+    Objects.requireNonNull(title);
+
+    log.info("Executing reports");
+    List<AuditorResults> results = reporters.values().parallelStream().map(reporter -> {
+      return reporter.get();
+    }).flatMap(List::stream).collect(Collectors.toList());
+
+    log.info("Generating page head and table of contents");
+    sink.head();
+    sink.title();
+    sink.text(title);
+    sink.title_();
+    sink.head_();
+
+    sink.body();
+    sink.section1();
+    sink.text("Table of Contents");
+    sink.section1_();
+
+    sink.list();
+    results.stream().forEach(result -> {
+      sink.listItem();
+      /*
+       * prepend "id-" to result.getId() because UUIDs can start with numbers, but
+       * HTML anchors can't
+       */
+      sink.link(String.format("#id-%s", result.getId()));
+      sink.text(result.getName());
+      sink.link_();
+      sink.listItem_();
+    });
+    sink.list_();
+
+    log.info(String.format("Beginning report sections generation (%d reporters found)", reporters.size()));
+    results.stream().forEach(result -> {
+      log.info(String.format("Found audit result named %s", result.getName()));
+      try {
+        createSection(result, sink);
+      } catch (ParseException e) {
+        throw new RuntimeException(e);
+      }
+    });
+    sink.body_();
+    log.info("Generation complete");
+  }
+
+  private void createSection(AuditorResults results, Sink sink) throws ParseException {
+    Reader confidentiality = new StringReader(results.getConfidentialityStatement());
+    Reader intro = new StringReader(results.getIntroduction());
+    Parser p = new MarkdownParser();
+
+    sink.anchor(String.format("id-%s", results.getId()));
+    sink.anchor_();
+    sink.section1();
+
+    sink.sectionTitle1();
+    sink.text(results.getName());
+    sink.sectionTitle1_();
+
+    sink.paragraph();
+    p.parse(confidentiality, sink);
+    sink.paragraph_();
+
+    sink.paragraph();
+    p.parse(intro, sink);
+    sink.paragraph_();
+
+    long duration = Duration.between(results.getTimestampStart().toInstant(), results.getTimestampEnd().toInstant())
+        .toMillis();
+    resultMetrics(results.getDescriptionHeaders(), results.getResults(), duration, sink);
```

Comment:
`Duration` is an available type.  Mebbe `AuditResults` has a `Duration getDuration()`?

Reply:
Counterpoint:

```Caused by: org.codehaus.modello.model.ModelValidationException: Field 'duration': Illegal type: 'java.time.Duration'.
    at org.codehaus.modello.model.ModelField.validateElement (ModelField.java:224)
```

Reply:
Sorry.  `Duration` is an available *Java* type.